samcli/lib/build/app_builder.py

```diff
@@ -0,0 +1,305 @@
+"""
+Builds the application
+"""
+
+import os
+import io
+import json
+import logging
+from collections import namedtuple
+
+try:
+    import pathlib
+except ImportError:
+    import pathlib2 as pathlib
+
+import docker
+
+import samcli.lib.utils.osutils as osutils
+from samcli.local.docker.lambda_build_container import LambdaBuildContainer
+from aws_lambda_builders.builder import LambdaBuilder
+from aws_lambda_builders.exceptions import LambdaBuilderError
+from aws_lambda_builders import RPC_PROTOCOL_VERSION as lambda_builders_protocol_version
+
+
+LOG = logging.getLogger(__name__)
+
+
+class UnsupportedRuntimeException(Exception):
```

Comment:
nit: these exceptions could go into its own file

Reply:
Agreed. I started with something small and expanded in scope. We should move them out later